### DISCUSSION TITLE
How to "getFont" (read currently used font)

### DISCUSSION THREAD

**Discussion by venice1200** [olikraus/u8g2#1387] [Q&A] [answered]
Hi,
I am using Arduino 1.8.13.

I like to store the currently used font temporarily in an variable, 
change the font, do something with the new font 
and set the font back to the stored one a few lines later.

Is the an getFont or another way to do this.
Many Thx for the cool library.
Lars

**Comment by olikraus** [comment, marked as answer]
You can just declare a variable like this:
```
const uint8_t *font;
```
Then assign any font to this variable and use the variable as argument for setFont.

**Reply by venice1200** [reply]
Thx for answering.
There is no function like *getFont* to store the result in your declared variable?
I like to „read“ the actual used font.

**Reply by olikraus** [reply]
```
uint8_t *font = u8g2.getU8g2()->font;
```

**Reply by venice1200** [reply]
Perfect, I will try this later.
Many thanks.

**Reply by venice1200** [reply]
Is this correct as I am not a specialist in Pointer & Co.

---In Loop---
// Save font
const uint8_t *old_font = u8g2.getU8g2()->font;

// Set new font
u8g2.setFont(u8g2_font_maniac_tr);
// Do something....

// Set font back
u8g2.setFont(old_font);

It works fine and I don't get compiler errors or warnings.

**Reply by olikraus** [reply]
It looks correct. And yes, it this works, then great.
